### PR TITLE
New keyword: single-arg

### PR DESCRIPTION
**Pull request by pciarach**
Introduced `single-arg` keyword to allow passing comma-separated CSS values as mixins' arguments

### CODE REVIEW

**Comment by pciarach** [comment]
Hi @ai, I've created PR as we discussed in #161.

As I was always using mixins defined inside CSS files so I'm not super experienced with the other options and possible cross-interactions. I've tried to include some of them in the test cases but let me know if you can think of any others.

File: README.md [hunk 1]
```diff
@@ -257,6 +257,21 @@ could be used like this:
 .isIE .foo { color: red; }
 ```
 
+### Comma-separated CSS values
+
+In order to pass a comma-separated value as a argument to a mixin, you can use
+the special `asSingleArg` keyword. For example:
```
Comment (pciarach) [inline README.md:263]: Other names that I came up with while coding:
- `single-arg`
- `mixinArg`
- `mixin-arg`

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,4 @@
 node_modules/
 
 coverage/
+.idea/
```
Comment (ai) [inline .gitignore:4]: IDE’s files should be in user’s local ignore file `core.excludesFile`

File: README.md [hunk 3]
```diff
@@ -257,6 +257,21 @@ could be used like this:
...
+
+```css
+@define-mixin transition $properties, $duration {
+  transition-property: $properties;
+  transition-duration: $duration;
+}
+
+.foo {
+    @mixin transition asSingleArg(color, background-color), 0.5s;
```
Comment (ai) [inline README.md:272]:
```suggestion
  @mixin transition asSingleArg(color, background-color), 0.5s;
```

File: index.js [hunk 4]
```diff
@@ -8,6 +8,7 @@ let glob = require('fast-glob')
 
 let MIXINS_GLOB = '*.{js,json,css,sss,pcss}'
 let IS_WIN = platform().includes('win32')
+let SINGLE_ARGUMENT_KEYWORD = 'asSingleArg'
```
Comment (ai) [inline index.js:11]: Just inline string in the code. Variable is longer and we will not change it a lot.

File: package.json [hunk 5]
```diff
@@ -53,6 +53,7 @@
     "eslint-plugin-prefer-let": "^3.0.1",
     "eslint-plugin-promise": "^6.1.1",
     "postcss": "^8.4.31",
+    "prettier": "^3.2.5",
```
Comment (ai) [inline package.json:56]: Prettier is 8 MB in `node_modules`. I have about 30 projects on my machine, 8 MB is too much.

Just use Prettier from your IDE.

File: index.js [hunk 6]
```diff
@@ -111,15 +112,56 @@ function processMixinContent(rule, from) {
...
+        break
+    }
+  }
+}
+
+function resolveSingleArgumentValue(value) {
+  let content = value.slice(SINGLE_ARGUMENT_KEYWORD.length).trim()
+
+  if (!content.startsWith('(') || !content.endsWith(')')) {
+    throw new Error(
```
Comment (ai) [inline index.js:154]: This error will not be connected with CSS line and column.

Use `Node#error` instead.

File: README.md (hunk 1 above)
Comment (ai) [reply README.md:263]: I think CSS code style is more about `single-arg`

camelCase is very rare.

**Comment by ai** [comment]
Thanks. Released in 10.0.

**Comment by pciarach** [comment]
Thank you for the help and review!

**Pull request merged**